test/framework/src/main/java/org/elasticsearch/repositories/blobstore/ESBlobStoreRepositoryIntegTestCase.java

```diff
@@ -91,8 +91,13 @@ protected Settings repositorySettings(String repoName) {
         return Settings.builder().put("compress", randomBoolean()).build();
     }
 
+    /**
+     * Create a repository without verification, assuming the repository is perfectly valid.
+     * If the repository can be invalid, use another method
+     * {@link ESBlobStoreRepositoryIntegTestCase#createRepository(String repoName, boolean verify)}
+     */
     protected final String createRepository(final String name) {
-        return createRepository(name, true);
+        return createRepository(name, false);
```

Comment:
Why is this change necessary? Is it because we will reject invalid repo earlier with this change and we still want some invalid repo to be created in tests? Sounds like an exceptional use case. Are there many such usages? If not, I feel it might be better for clarity to have a separate method such as `createRepositoryWithoutVerification` for them. The following method `String createRepository(final String name, final boolean verify)` is only used here and probably can be dropped.

Reply:
There are several tests that dont rely on verification, repository creation is a part of the setup but no assertions done. In some tests we create a snapshot thread pool with single thread, new verification code also uses snapshot thread and it causes test to hang. My understanding it happens when we intentionally block some repository calls though mocks. At least thats what I was able to reproduce with debugger.

Also not all tests I was able to catch locally, some of them fail only on CI. Even not on my GCP instance.

I can try to refactor with `createRepositoryWithoutVerification` for clarity.

Reply:
> In some tests we create a snapshot thread pool with single thread, new verification code also uses snapshot thread and it causes test to hang. 

The existing code also performs verification in the snapshot thread pool after the repo is added to the cluster state. So it is not immediately clear to me why the new verficiation step would hang?

Reply:
Let me reproduce these and post details here.

Reply:
I might mess up a bit here, this change is not related to hanging tests. 

Another diff in this PR addresses hanging test:
`RepositoriesIT -> testRepositoryConflict`
```java

        blockMasterOnWriteIndexFile(repo);
        logger.info("--> start deletion of snapshot");
        ActionFuture<AcknowledgedResponse> future = clusterAdmin().prepareDeleteSnapshot(repo, snapshot1).execute();
        logger.info("--> waiting for block to kick in on node [{}]", blockedNode);
        waitForBlock(blockedNode, repo);

        ...

        logger.info("--> try updating the repository, should fail because the deletion of the snapshot is in progress");
        RepositoryConflictException e2 = expectThrows(
            RepositoryConflictException.class,
            clusterAdmin().preparePutRepository(repo)
                .setVerify(true) // >>> will hang on verification
                .setType("mock")
                .setSettings(Settings.builder().put("location", randomRepoPath()))
        );

        ...

        logger.info("--> unblocking blocked node [{}]", blockedNode);
        unblockNode(repo, blockedNode); // >>> release thread here
```

So here we use thread pool with single thread that is busy with snapshot deletion, then we issue another call to repository service and it hangs on verification. Test unblocks node after.

Reply:
This change related to metrics. We have tests that verify we have exact Repository metrics - get/put/list across nodes. But with new change master node will have few more, and there are several tests fail. Not consistently thou.

https://gradle-enterprise.elastic.co/s/w4amqdgguyipy